src/common/src/types/mod.rs

```diff
@@ -341,8 +341,11 @@ impl DataType {
         };
         match self {
             DataType::Struct(t) => {
+                if !t.is_unnamed() {
+                    // To be consistent with `from_protobuf`, we only set field names when it's a named struct.
```

Comment:
nit: I didn't found the `from_protobuf` func
```suggestion
                    // To be consistent with `From<&PbDataType>`, we only set field names when it's a named struct.
```